core/execution/future/market.go

```diff
@@ -357,11 +358,12 @@ func (m *Market) OnEpochEvent(ctx context.Context, epoch types.Epoch) {
 	case vegapb.EpochAction_EPOCH_ACTION_START:
 		m.liquidity.UpdateSLAParameters(m.mkt.LiquiditySLAParams)
 		m.liquidity.OnEpochStart(ctx, m.timeService.GetTimeNow(), m.markPrice, m.midPrice(), m.getTargetStake(), m.positionFactor)
+		m.epoch = epoch
```

Comment:
Can you also set `m.epoch` in `OnEpochRestore()`? Its used when a node restores from a snapshot, so we can make sure this field is set back again.